tracing-undertow/src/main/java/com/palantir/tracing/undertow/TracingAttachments.java

```diff
@@ -27,5 +31,17 @@ public final class TracingAttachments {
     /** Attachment providing the request identifier. */
     public static final AttachmentKey<String> REQUEST_ID = AttachmentKey.create(String.class);
 
+    /**
+     * Detached span object representing the entire request including asynchronous components.
+     * This is intentionally not public, we expose only the {@link Detached} component which critically does not
```

Comment:
> we expose only the {@link Detached} component which critically does not
     * include {@link DetachedSpan#complete()} APIs

I assume this is important so that the caller can only observe the span, not inadvertently call `complete()` which is handled by the exchange completion handler?

Reply:
Yep, the idea is that when we create this span, we register the exchange completion listener to close it -- we don't want user code to inadvertently close the span _before_ the request has a chance to complete. This object can be used to create new child spans, which themselves may be closed at the users discression.